coregistration/eolearn/tests/test_coregistration.py

```diff
@@ -11,54 +11,51 @@
+from eolearn.coregistration import ECCRegistrationTask
 
 logging.basicConfig(level=logging.DEBUG)
 
 
-@pytest.fixture(name="eopatch")
-def eopatch_fixture():
-    bands = np.zeros((2, 20, 20, 1))
-    bands[1] = np.arange(400).reshape(1, 20, 20, 1) / 400
-    bands[0] = bands[1]
-    bands[1, 5:15, 5:15, :] = 0.5
-    bands[0, 7:17, 5:15, :] = 0.5
-    mask = np.ones((2, 20, 20, 1), dtype=np.int16)
-    ndvi = np.ones((2, 20, 20, 1))
-    dem = np.ones((20, 20, 1))
+def test_registration(example_eopatch):
+    apply_to_features = [(FeatureType.DATA, "NDVI"), (FeatureType.DATA, "BANDS-S2-L1C"), (FeatureType.MASK, "CLM")]
 
-    eop = EOPatch()
-    eop[FeatureType.DATA, "bands"] = bands
-    eop[FeatureType.DATA, "ndvi"] = ndvi
-    eop[FeatureType.MASK, "cm"] = mask
-    eop[FeatureType.DATA_TIMELESS, "dem"] = dem
-    return eop
-
-
-def test_registration(eopatch):
     reg = ECCRegistrationTask(
-        (FeatureType.DATA, "bands"),
-        valid_mask_feature=(FeatureType.MASK, "cm"),
-        interpolation_type=InterpolationType.NEAREST,
-        apply_to_features={FeatureType.DATA: ["bands", "ndvi"], FeatureType.MASK: ["cm"]},
+        registration_feature=(FeatureType.DATA, "NDVI"),
+        reference_feature=(FeatureType.DATA_TIMELESS, "MAX_NDVI"),
+        channel=0,
+        valid_mask_feature=None,
+        apply_to_features=apply_to_features,
+        interpolation_mode=cv2.INTER_LINEAR,
+        warp_mode=cv2.MOTION_TRANSLATION,
+        max_iter=100,
+        gauss_kernel_size=1,
+        border_mode=cv2.BORDER_REPLICATE,
+        border_value=0,
+        num_threads=-1,
+        max_translation=5.0,
     )
     with pytest.warns(EORuntimeWarning):
-        reopatch = reg(eopatch)
+        reopatch = reg(example_eopatch)
 
-    assert eopatch.data["bands"].shape == reopatch.data["bands"].shape, "Shapes of .data['bands'] do not match"
-    assert eopatch.data["ndvi"].shape == reopatch.data["ndvi"].shape, "Shapes of .data['ndvi'] do not match"
-    assert eopatch.mask["cm"].shape == reopatch.mask["cm"].shape, "Shapes of .mask['cm'] do not match"
     assert (
-        eopatch.data_timeless["dem"].shape == reopatch.data_timeless["dem"].shape
+        example_eopatch.data["BANDS-S2-L1C"].shape == reopatch.data["BANDS-S2-L1C"].shape
     ), "Shapes of .data['bands'] do not match"
-    assert not np.allclose(eopatch.data["bands"], reopatch.data["bands"]), "Registration did not warp .data['bands']"
-    assert not np.allclose(eopatch.data["ndvi"], reopatch.data["ndvi"]), "Registration did not warp .data['ndvi']"
-    assert not np.allclose(eopatch.mask["cm"], reopatch.mask["cm"]), "Registration did not warp .mask['cm']"
-    assert np.allclose(
-        eopatch.data_timeless["dem"], reopatch.data_timeless["dem"]
-    ), "Registration did warp data_timeless"
+    assert example_eopatch.data["NDVI"].shape == reopatch.data["NDVI"].shape, "Shapes of .data['ndvi'] do not match"
```

Comment:
i think you can remove the text messages of the assert, since it is clear from the statement what the issue is, and pytest does a great job of displaying stuff.

I only mean the ones in L49, L50, and L58, the rest are informative and I'd keep them

Reply:
:+1: removed